### PR TITLE
Run PRs in Concourse

### PR DESCRIPTION
**Pull request by cybojenix**
This sets up the PR pipeline, and part of the deployment pipeline for Concourse.

The PR pipeline is currently running on the GDS concourse, as well as local concourse environments.

While a deploy pipeline does exist, it currently is not set up to run yet.
This will occur at a later time.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,30 @@
 DOCKER_COMPOSE = docker-compose -f docker-compose.yml
 BUNDLE_FLAGS=
+
 ifdef DEPLOYMENT
   BUNDLE_FLAGS = --without test development
 endif
 
 ifndef JENKINS_URL
```
Comment (emileswarts) [inline Makefile]: Should we just remove all references to Jenkins?
Are we keeping this backwards compatible as we make the switch to concourse?

File: ci/pipelines/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+resources:
+  - name: govwifi-admin
+    type: git
+    source:
+      uri: "https://github.com/alphagov/govwifi-admin.git"
+      branch: "concourse"
+
+  - name: ecr
+    # This needs to be docker-image, until registry-image supports ECR
```
Comment (emileswarts) [inline ci/pipelines/deploy.yml]: Not entirely sure what this comments means.
ECR isn't a supported registry?

Comment (cybojenix) [reply ci/pipelines/deploy.yml]: https://github.com/concourse/registry-image-resource/pull/27

feature is being worked on.
Concourse recommends `registry-image` over `docker-image`

File: Makefile (hunk 1 above)
Comment (cybojenix) [reply Makefile]: leaving them in to make the switch easier

**Comment by tommotaylor** [comment]
Looks good to me. Will wait for @emileswarts to approve

File: ci/pipelines/deploy.yml (hunk 2 above)
Comment (dan-hill2802) [reply ci/pipelines/deploy.yml]: Hoping to get it merged soon. In the meantime you could use:
```
- name: registry-image-resource
  type: docker-image
  source:
    repository: dwpdigital/registry-image-resource
```

Comment (cybojenix) [reply ci/pipelines/deploy.yml]: Thanks for that @dan-hill2802. 
For now, we'll stick with using `docker-image` until the functionality is mainlined.
There's no pressing need to use `registry-image`

**Pull request merged**